src/zap/zapmetadata.cpp

```diff
@@ -171,8 +172,8 @@ void ZapILMetaData::CopyRVAFields()
         // Handle overlapping fields by reusing blobs based on the address, and just updating size and alignment.
```

Comment:
Is this comment still valid?

Reply:
It looks valid, but that means that `_ASSERTE(!pRVADataNode->IsPlaced());`a few lines below is incorrect.

Try to create a repro in IL that has two fields pointing to the same RVA.

Reply:
Hmm, good point. I don't know if there will be a real world scenario where this happens, but I agree the assert is too aggressive here.
Here's what I'll do: I'll add a debug-only loop after the qsort where I check that no node is placed. I want to have an assert in case someone makes a change that causes some nodes to be placed out of order. It was a bit of a headache to diagnose this...